contracts/SwitcheoTokenZIL.scilla

```diff
@@ -47,30 +55,34 @@ let get_val =
   | None => zero
   end
 
+let some_addr = 
+  fun (addr: ByStr20) =>
+  Some {ByStr20} addr
+
 (***************************************************)
 (*             The contract definition             *)
 (***************************************************)
 
 contract FungibleToken
 (
-  contract_owner: ByStr20,
-  name : String,
+  name: String,
   symbol: String,
   decimals: Uint32,
-  init_supply : Uint128
+  init_supply: Uint128,
+  init_minter: ByStr20
```

Comment:
should this be `init_lockproxy`?

minter is right now playing the role of minting and burning

Reply:
Yes, minter is lockproxy contract. but it is okay to keep the name right?

Reply:
to me, it was a bit confusing as minter does minting and burning at the same time

Reply:
okie